flashinfer/jit/core.py

```
@@ -94,18 +94,64 @@ def _print_once(self, log_method, msg: str, *args: Hashable) -> None:
+
+    # Use compilation context (from FLASHINFER_CUDA_ARCH_LIST or auto-detect at import time)
+    archs = current_compilation_context.TARGET_CUDA_ARCHS
+    eligible = _archs_eligible(archs)
+
+    # If context was empty (e.g. CUDA not visible at first import), re-query once
+    if not eligible and len(archs) == 0:
+        try:
+            fresh = CompilationContext()
+            if fresh.TARGET_CUDA_ARCHS:
+                eligible = _archs_eligible(fresh.TARGET_CUDA_ARCHS)
+        except Exception as e:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)